File: 9_remove-element/solution.cpp
#include <vector>
#include <iostream>

int removeElement(std::vector<int>& nums, int val) {
    int counter = 0;
    for (int i = 0; i < nums.size(); i++) {
        if (nums[i] != val) {
            nums[counter] = nums[i];
            counter++;
        }
    }
    return counter ;
}

int main() {
    std::vector<int> nums = {0,1,2, 2, 3,0, 4, 2};
    int val = 2;
    int counter = removeElement(nums, val);
    std::cout << val << std::endl;
    return 0;
}

File: 9_remove-element/solution.py
def removeElement(nums, val):
    """
    :type nums: List[int]
    :type val: int
    :rtype: int
    """
    counter = 0
    for i in range(len(nums)):
        if nums[i] != val:
            nums[counter] = nums[i]
            counter += 1
    
    return counter , nums

nums = [0,1,2,2,3,0,4,2]
val = 2

counter , nums = removeElement(nums,val)

print(counter , nums)


    